src/pages/benchviz/Benchmark.tsx

```diff
@@ -0,0 +1,151 @@
+    benchmarkName: string;
+}
+
+// Utility functions
+const formatBenchmarkName = (benchmarkName: string) => benchmarkName.replace(".lua", "").split("/").pop()!;
+const formatMemory = (value: number) => `${Math.round(value / 10) / 100} Kb`;
+
+const benchmarkGarbage = (bm: BenchmarkResult) => bm.categories[MemoryBenchmarkCategory.Garbage];
+const garbagePercentChange = (result: JoinResult<BenchmarkResult>) =>
+    (benchmarkGarbage(result.right!) - benchmarkGarbage(result.left!)) / benchmarkGarbage(result.left!);
+
+// Get benchmark data from URL parameter
+const data = window.location.search.split("?d=")[1];
+const results = JSON.parse(zlib.inflateSync(Buffer.from(data, "base64")).toString());
+
+const dataMaster = results.old as BenchmarkResult[];
+const dataCommit = results.new as BenchmarkResult[];
+
+// Match old/new results by name
+const matchedBenchmarks = joinOnProperty(dataMaster, dataCommit, (bm) => bm.benchmarkName);
+
+// Sort by percentage change of garbage created
+const benchmarksSortedByPercentDifference = matchedBenchmarks.sort(
+    (a, b) => garbagePercentChange(a) - garbagePercentChange(b),
+);
+
+// Table containing the benchmark results
+const resultsTable: JSX.Element[] = [];
+benchmarksSortedByPercentDifference.forEach((bm, i) => {
+    const change = garbagePercentChange(bm);
+    const rowColor = change === 0 ? "currentColor" : change > 0 ? "red" : "green";
+
+    resultsTable.push(
+        <tr key={i} style={{ color: rowColor }}>
+            <td>{bm.left?.benchmarkName}</td>
+            <td>{formatMemory(benchmarkGarbage(bm.left!))}</td>
+            <td>{formatMemory(benchmarkGarbage(bm.right!))}</td>
+            <td>{change}</td>
+        </tr>,
+    );
+});
+
+// Comparison data master garbage created vs commit garbate created (ABSOLUTE)
+const generatedGarbageData = benchmarksSortedByPercentDifference.map((bm) => ({
+    name: formatBenchmarkName(bm.left?.benchmarkName ?? bm.right?.benchmarkName!),
+    oldValue: bm.left?.categories[MemoryBenchmarkCategory.Garbage] ?? 0,
+    newValue: bm.right?.categories[MemoryBenchmarkCategory.Garbage] ?? 0,
+}));
+
+let garbageCreatedComparisonSvg: SVGSVGElement;
+const garbageCreatedComparisonGraphWidth = 1000;
+const garbageCreatedComparisonGraphHeight = 300;
+
+// Comparison data master garbage created vs commit garbate created (ABSOLUTE)
+const generatedGarbageChangeData = matchedBenchmarks.map((bm) => {
+    const oldValue = bm.left?.categories[MemoryBenchmarkCategory.Garbage]!;
+    const newValue = bm.right?.categories[MemoryBenchmarkCategory.Garbage]!;
+
+    return {
+        name: formatBenchmarkName(bm.left?.benchmarkName ?? bm.right?.benchmarkName!),
+        value: (100 * (newValue - oldValue)) / oldValue,
+    };
+});
+
+let garbageCreatedChangeSvg: SVGSVGElement;
+const garbageCreatedChangeGraphWidth = 1000;
+const garbageCreatedChangeGraphHeight = 300;
+
+export default class extends React.Component {
```

Comment:
Class components is pretty much an anti-pattern in react at this point. This component would be quite simple with hooks:

```ts
export default function Benchmark() {
    useEffect(() => {
        // contents of draw()
    });

    return (
        // ...
    );
}
```